### PR TITLE
proposal: Basic repository structure

### PR DESCRIPTION
**Pull request by GabeMillikan**
This PR just creates the basic structure of the repository. _**I'm open to suggestions!**_ This is just my proposal

I've decided to make a few changes from the way that Pax-Academia is setup:

1. **Use `discord.py` instead of `pycord`**\
    because `discord.py` has received some massive updates "recently" and it is now a complete implementation of the Discord API. It supports slash commands and all the other "recent" goodies. There's not really a good reason to use `pycord` anymore.
2. **Use `discord.Client` instead of `discord.ext.commands.Bot`**
    because we won't have any prefix commands. We have no purpose for the extra bloat of `discord.ext`, so just get rid of it.
3. **Don't use `discord.ext.commands.Cog`**
    because they are a.) uselessly complicated (why not just use regular classes?) and b.) not compatible with `discord.Client`

Overall, we end up with the following structure:
```py
Pax-Virtuoso/
├── abilities/          # contains everything that the bot "does"
│   ├── __init__.py     # simply imports all of the abilities
│   ├── music.py        # music playing/queueing etc. functionality
│   ├── meta.py         # stuff relating to the bot itself (/ping, /view-logs, etc.)
│   └── TODO.py         # we may add transcription or other VC functionalities?
├── bot.py              # just initializes the bot and command tree, so that it's easily accessible throughout the repository
├── config.py           # basically just loads the .env and validates the variables
└── main.py             # calls bot.run(token)    
```

and slash commands are pretty self explanatory:
```py
import discord
from bot import client, tree

@tree.command(description="Checks the bot's ping time")
async def ping(interaction: discord.Interaction) -> None:
    await interaction.response.send_message(
        f"Ping! `{client.latency * 1000:.2f} milliseconds` "
    )
```

SIDE NOTE:
If we ever need to add a database, I would really like to try using SQLAlchemy, mostly just because I've never used it before and I'd really like to learn it.

### CODE REVIEW

File: ENVIRONMENT.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+│   ├── __init__.py
+│   └── music.py
+├── bot.py
+├── main.py
+├── README.md
+└── .env               <--- HERE
+```
+then, open the file in a text editor and fill out the required variables in the following format (replacing the example information with your own):
+```
+API_TOKEN=ASJD3KLFaLLsKSADFJ23KLSFD.ASkhjsfd.ASHJKDaADKSLJ_ASDsH12JH90JlpPASM23
```
Comment (GabeMillikan) [inline ENVIRONMENT.md:19]: note: this is just keyboard mash, not a real token haha

File: abilities/meta.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from discord import Interaction
+
+from bot import client, tree
+
+
+@tree.command(description="Get the bot's ping time to Discord")
+async def ping(interaction: Interaction) -> None:
+    await interaction.response.send_message(
+        f"Pong! My ping time to Discord is {client.latency * 1000:.0f} milliseconds."
```
Comment (danparizher) [inline abilities/meta.py]:
```suggestion
        f"Pong! My ping time to Discord is {client.latency * 1000:.0f} milliseconds.",
```

File: abilities/music.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+
+@tree.command(description="Plays a song")
+@app_commands.describe(name="The name of the song.")
+async def play(interaction: Interaction, name: str) -> None:
+    """
+    TODO: Implement & document this command
+    """
+    await interaction.response.send_message(
+        f"Hello {interaction.user.mention}! Now playing: {name}"
```
Comment (danparizher) [inline abilities/music.py]:
```suggestion
        f"Hello {interaction.user.mention}! Now playing: {name}",
```

File: bot.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+# so that when someone plays a song, we can check what voice channel they're in
+intents.voice_states = True
+
+client = discord.Client(intents=intents)
+tree = discord.app_commands.CommandTree(client)
+
+
+@client.event
+async def on_ready():
```
Comment (danparizher) [inline bot.py]:
```suggestion
async def on_ready() -> None:
```

File: config.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+def getenv_string(environment_variable: str) -> str:
+    s = os.getenv(environment_variable)
+
+    if not s:
+        raise Exception(
+            f"Required environment variable {environment_variable!r} is not set. Please read ./ENVIRONMENT.md"
+        )
+
+    return s
```
Comment (danparizher) [inline config.py]:
```suggestion
    if s := os.getenv(environment_variable):
        return s
    msg = f"Required environment variable {environment_variable!r} is not set. Please read ./ENVIRONMENT.md"
    raise OSError(msg)
```
Suggesting this for a few reasons:
1. We want to use named expressions to simplify assignment and conditionals
2. Swapping the if-else branches
3. Assigning exception to a variable. Python includes the `raise` in the default traceback. By using an f-string, the error message will be duplicated in the traceback, which can make the traceback less readable.
4. Specifying OSError

**Review by danparizher — CHANGES_REQUESTED** [review]

File: config.py (hunk 5 above)
Comment (GabeMillikan) [reply config.py]: All sounds good to me, I've never thought about number 3 before, very good point!

**Review by danparizher — APPROVED** [review]

**Pull request merged**